### ISSUE TITLE
Ensure all prometheus metrics are available from Azure AKS benchmark

### ISSUE THREAD

**Issue by danmason** [xtdb/crux#3701]
Want to add whatever config/setup we need such that we can log out all of our relevant prometheus metrics from the Azure AKS benchmark.
- [x] Should write up any relevant instructions necessary for setting it up.
- [x] Want to ensure we can distinguish between nodes - probably want to look at "dimensions" for this.
- [x] Auctionmark benchmark gauges available.
- [x] JVM gauges available.
- [x] Setup a basic dashboard of things we care about in debugging the benchmark.

**Comment by tggreene** [comment]
Currently we have an application insights instance that we send telemetry to, this allows us to use the monitoring > metrics option and then select the metric namespace azure.applicationinsights with xt and benchmark metrics.

There's also a dashboard (just search dashboard in the portal) `cloud-benchmark` which shows a handful of metrics.

Currently it's not configured to handle multiple reporters (the three nodes of our multi node benchmark).

**Comment by danmason** [comment]
From @jarohen comments on our planning meeting - might be worth a look into prometheus "dimensions" (and how they may translate/apply on AKS/applicationinsights monitoring) for handling multiple nodes: https://prometheus.io/docs/concepts/data_model/

**Comment by tggreene** [comment]
From slack from me:
> I think for monitoring it’s going to be best to proceed with [application insights instance per node](https://learn.microsoft.com/en-us/azure/architecture/guide/multitenant/service/application-insights#one-application-insights-instance-per-stamp); we can add properties to metrics via the azure sdk but there’s no obvious way to even filter by these in the monitoring/metrics interface

**Comment by tggreene** [comment]
We're trying adding a tag via micrometer to see if this can be selected within Azure.

**Comment by danmason** [comment]
Added a shared common tag on our metrics registry, "node-id". Currently just generating names for this, suffixed with a UUID, and tagging in the registry startup. We could potentially allow this to be passed in as a configuration option to the registry, such that we could easily tell which node is which. 

The prometheus tags WERE reflected within azure, though to be able to act on we had to update one of the billing settings in application insights to allow for it, namely: "alert on custom metrics".

After allowing this, we have the ability to:
- Filter on custom metrics on tag.
- Split custom metrics on tag.

This allows us to see node level metrics.

Currently writing up an auctionmark dashboard for us to get info from auctionmark within application insights:
![image](https://github.com/user-attachments/assets/5a3c181c-f90b-4c88-b1a4-51dee3652870)
